python/pfs/drp/stella/reduceArc.py

```diff
@@ -199,6 +206,9 @@ def gather(self, dataRefList, results):
         if len(results) == 0:
             raise RuntimeError("No input spectra")
 
+        if not self.config.doUpdateDetectorMap:
+            self.log.fatal("you can't get here")
```

Comment:
While this should never happen, a more meaningful message would be helpful, eg., "Something's wrong: code is creating a new detectorMap although doUpdateDetectorMap=False"

Reply:
An `assert` or `raise AssertionError` might be appropriate.